plugin/action/add_host/add_host.go

```diff
@@ -39,15 +41,18 @@ func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
 
 func (p *Plugin) Start(config pipeline.AnyConfig, _ *pipeline.ActionPluginParams) {
 	p.config = config.(*Config)
+
+	var err error
+	p.hostname, err = os.Hostname()
+	if err != nil {
+		logger.Fatal(err)
+	}
 }
 
 func (p *Plugin) Stop() {
 }
 
 func (p *Plugin) Do(event *pipeline.Event) pipeline.ActionResult {
-	hostname, _ := os.Hostname()
-
-	event.Root.AddFieldNoAlloc(event.Root, p.config.Field).MutateToString(hostname)
-
+	event.Root.AddFieldNoAlloc(event.Root, p.config.Field).MutateToString(p.hostname)
```

Comment:
Let's make a copy of the `p.hostname`. This is necessary because we can change the bytes inside the node, so the hostname will also change.

I wrote a simple example to demonstrate this:
https://go.dev/play/p/R5AzBtBCPfp

Reply:
@vitkovskii, should we handle such cases?